include/finufft/finufft_core.h

```diff
@@ -138,7 +149,8 @@ template<typename TF> struct FINUFFT_PLAN_T { // the main plan object, fully C++
 
   // These default and delete specifications just state the obvious,
   // but are here to silence compiler warnings.
-  FINUFFT_PLAN_T() = default;
+  FINUFFT_PLAN_T(int type, int dim, const BIGINT *n_modes, int iflag, int ntrans, TF tol,
```

Comment:
This is a bit of a mess. I had to add the `ier` reference here to report the warning state ("epsilon too small") properly. "Real" error conditions are handled by exceptions.